phone_verify/backends/base.py

```diff
@@ -132,10 +132,13 @@ def validate_security_code(self, security_code, phone_number, session_token):
             return stored_verification, self.SECURITY_CODE_VERIFIED
 
         # mark security_code as verified
-        stored_verification = SMSVerification.objects.filter(
+        stored_verification = SMSVerification.objects.get(
```

Comment:
Virtually these APIs can fail. For example in cases when no `SMSVerification` object is found, or maybe when multiple objects match the criteria.

It is important here to handle these cases.

Reply:
Ok I will handle those cases. I just need to return `None` in case the instance can't be retrieved, right?

Reply:
On a second thought, while going through the code, I think it is better to avoid this DB call altogether and just use `stored_verification` object that was retrieved and tested earlier.

What do you think?

Reply:
Yes, that would be a much better solution and no need to handle any corner cases either!

Just to make sure we are on the same page, you are suggesting doing this instead

This is what was retrieved:
```
        stored_verification = SMSVerification.objects.filter(
            security_code=security_code, phone_number=phone_number
        ).first()
```
I would suggest replacing that with:

```
        stored_verification = SMSVerification.objects.filter(
            security_code=security_code, phone_number=phone_number, session_token=session_token
        ).first()
```
The reason is that when all 3 are queried together we can be sure it would return a unique instance, if it returns anything at all. This may not happen if the session token is omitted because a user may sign up using their number etc and may forget that they have used their number in the past and try to sign up again with the same number. 

That could create multiple instances and throw an error since `SMSVerification phonefield`  has to be unique but session token would always be different. With that being said,  I haven't gone through the entire code so not sure if this scenario is being handled or not.

And then my commit could look like this:
```

        stored_verification.is_verified = True
        stored_verification.save()

```

I have already verified that these changes would work: my suggestion and updating the instance field like you suggested!

What do you think?

Reply:
Hi @arnav13081994 

I don't think that is a good idea simply because we won't be able to know if the `session_code` is not valid anymore (refer https://github.com/CuriousLearner/django-phone-verify/pull/29/files#diff-9f924f5888b070e5444e4db988177019R121)

`sesssion_code` is there so that the device that made the initial request can be verified as making the subsequent request.

So, you'll just need to use the same object and make the `is_verified` attribute `True` and save it to emit the `post_save` signal.

Please also make sure to update the release notes in README and some docs in docs folder about how others can use post_save signal to do some stuff.

Thanks!

Reply:
Ok good point. I'll do that.